mcs/class/test-helpers/RandomPort.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Linq;
+using System.Net.NetworkInformation;
+
+namespace MonoTests.Helpers {
+
+	public static partial class NetworkHelpers
+	{
+		public static int FindPort ()
+		{
+			// get list of current ports in use - this is vulnerable to race
+			// conditions, but with a large enough port range, the chances
+			// should be TINY
+			int[] current_ports = IPGlobalProperties.GetIPGlobalProperties ().GetActiveTcpConnections ().Select (x => x.LocalEndPoint.Port).ToArray ();
```

Comment:
I'm a bit worried that this will work correctly on all platforms. I googled a bit and the most recommded solution to get the next free port in .NET seems to be http://stackoverflow.com/questions/138043/find-the-next-tcp-port-in-net